addons/account/tests/test_reconciliation.py

```diff
@@ -2055,3 +2055,80 @@ def test_reconciliation_cash_basis_foreign_currency_low_values(self):
         pay_receivable_line1 = payment1.move_line_ids.filtered(lambda l: l.account_id == self.account_rcv)
         self.assertTrue(pay_receivable_line1.reconciled)
         self.assertEqual(pay_receivable_line1.matched_debit_ids, move_caba1.tax_cash_basis_rec_id)
+
+    def test_reconciliation_process_move_lines_with_mixed_currencies(self):
+        self.env['res.currency.rate'].create({
+            'currency_id': self.currency_usd_id,
+            'name': time.strftime('%Y') + '-01-01',
+            'rate': 2,
+        })
+
+        # Delete any old rate - to make sure that we use the ones we need.
+        old_rates = self.env['res.currency.rate'].search(
+            [('currency_id', '=', self.currency_usd_id)])
+        old_rates.unlink()
```

Comment:
aren't you deleting the one you just created with this? Should you not delete the rates and after that, create a new one?

Reply:
You are right. I didn't notice because right now the rate is not taken into consideration because the amount_currency is set.